src/main/java/redis/clients/jedis/ShardedJedisPool.java

```diff
@@ -83,12 +88,12 @@ public void destroyObject(PooledObject<ShardedJedis> pooledShardedJedis) throws
             if (!jedis.isBroken()) {
               jedis.quit();
             }
-          } catch (Exception e) {
+          } catch (RuntimeException e) {
```

Comment:
Exception was caught so that every node gets a chance to be destroyed even though there is a bug. This is an operation that MUST be done (as much as possible).

Reply:
Right now checked `Exception` is not thrown from the code inside - so my proposal is backward-compatible 
(I change Exception -> RuntimeException, not Throwable -> RutimeExceptio)

but catching generic checked `Exception` is quite "dangerous" if internal contract will change and something like `Interrupted Exception` will be thrown: current code will not handle it properly - but someone who make changes won't get compilation error and may miss this (as a result - thread won't be interrupted)

Reply:
> Exception was caught so that every node gets a chance to be destroyed even though there is a bug. This is an operation that MUST be done (as much as possible).

Probably I don't understood something what you mean, 
may you please explain some details how Exception->RuntimeException breaks current execution logic?

Reply:
@bokshitsky At lease separate these changes in another PR with explanation.